File: trace-web/trace/src/api/ApiCompanyInfo.ts
import { httpPost, httpGet, httpDelete, C_OK as _C_OK } from "./http";

export const C_OK = _C_OK;

export const add_company_info = async (params: any) => {
    return await httpPost("/trace-api/company_info/add_company_info", params);
};

export const delete_company_info = async (params: any) => {
    return await httpDelete("/trace-api/company_info/delete_company_info", params);
};

export const update_company_info = async (params: any) => {
    return await httpPost("/trace-api/company_info/update_company_info", params);
};

export const list_company_info = async (params: any) => {
    return await httpGet("/trace-api/company_info/list_company_info", params);
};

export const get_company_info = async (params: any) => {
    return await httpGet("/trace-api/company_info/get_company_info", params);
};

File: trace-api/src/api/api_company_info.py
#!/usr/bin/env python
# encoding: utf-8

from typing import Any
from fastapi import APIRouter
from ..obj.vobj_company_info import CompanyInfoObj
from ..obj.tobj_company_info import CompanyInfoForm
from ..obj.tobj_role import Perms
from ..obj import Resp, Page
from ..serv.serv_company_info import Server
from . import try_log

router = APIRouter()
server = Server()


@router.post("/add_company_info", summary="添加公司基本信息", response_model=Resp[Any])
@try_log(perm=Perms.company_info_edit)
async def add_company_info(form: CompanyInfoForm):
    return await server.add_company_info(form)


@router.delete("/delete_company_info", summary="删除公司基本信息", response_model=Resp[Any])
@try_log(perm=Perms.company_info_edit)
async def delete_company_info(id: int):
    return await server.delete_company_info(id)


@router.post("/update_company_info", summary="更新公司基本信息", response_model=Resp[Any])
@try_log(perm=Perms.company_info_edit)
async def update_company_info(form: CompanyInfoForm):
    return await server.update_company_info(form)


@router.get("/list_company_info", summary="查询公司基本信息列表", response_model=Resp[Page[CompanyInfoObj]])
@try_log(perm=Perms.company_info_view)
async def list_company_info(fuzzy: str = None, page_index: int = 0, page_size: int = 10):
    return await server.list_company_info(fuzzy=fuzzy, page_index=page_index, page_size=page_size)


@router.get("/get_company_info", summary="查询公司基本信息详情", response_model=Resp[CompanyInfoObj])
@try_log(perm=Perms.company_info_view)
async def get_company_info(id: int):
    return await server.get_company_info(id)
